flipperdevices/flipperzero-firmware
# Mifare Classic nonce log

# What's new

Some readers use different keys, depending on the tag's UUID. In these cases, Detect Reader's output is useless for key calculation with tools like [mfkey32v2](https://github.com/equipter/mfkey32v2), and it is necessary to use debug logs for this purpose. However, these logs require Flipper to be connected to another device during emulation. It is inconvenient in some scenarios.

This PR introduces a `MFC nonce log` toggle found in a new submenu NFC -> Settings. When enabled, nt/nr/ar received during Mifare Classic emulation will be written to `/int/.mfkey32_emul.log`. The toggle's state is stored in `/int/.nfc.settings`.

# Verification 

- Navigate to NFC -> Settings, set `MFC nonce log` to ON
- Emulate a Mifare Classic tag
- Read `/int/.mfkey32_emul.log`, received nonces should appear there
- Navigate to NFC -> Settings again, set `MFC nonce log` to OFF
- Emulate a Mifare Classic tag
- Read `/int/.mfkey32_emul.log`, nothing new should appear there

# Checklist (For Reviewer)
